src/session.ts

```diff
@@ -146,7 +150,13 @@ export const createState = (initial: State) => {
       store.update(s => {
         switch (s.connection) {
           case Connection.Connected:
-            s.session.address = address; 
+            // In case of locking Metamask the accountsChanged event returns undefined.
+            // To prevent out of sync state, the wallet gets disconnected.
+            if(address === undefined) disconnectWallet();
+            else {
+              s.session.address = address; 
+              window.localStorage.setItem("session", JSON.stringify({ address, tokenBalance: s.session.tokenBalance }))
```

Comment:
Should we also save the `tx` state?

Reply:
For persisting the session it has no direct value, but to comply with the interface type we should.
Eventually in the future, it'll be of use.